Gorilla/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"text/template"
+
+	"gorm.io/gorm"
+
+	"github.com/gorilla/mux"
+	_ "github.com/lib/pq"
+)
+
+var (
+	postgresDB *gorm.DB
+	tpl        *template.Template // Declare tpl here for global access
+)
+
+func main() {
+	var err error
+
+	/*---------------------
+	* Load env vars
+	*----------------------*/
+	dsn := "user=user password=password dbname=waect host=localhost port=5432 sslmode=disable" // Modify as needed
+
+	/*---------------------
+	 * Connect to DB
+	 *----------------------*/
+	postgresDB, err = connectDB(dsn) // Ensure this matches the function name in the db package
+	if err != nil {
+		log.Fatalf("Failed to connect to database: %v", err)
+	}
+
+	/*---------------------
+	* Setup routing
+	*----------------------*/
+	funcMap := template.FuncMap{
```

Comment:
It would be more clean if you create a function outside the main for setting up the routing and then call it directly in the template.new

```suggestion
	funcMap := template.FuncMap{
```
```suggestion
	tpl, err = template.New("timeline.html").Funcs(**NEWSETUPROTINGFUNCTIONHERE**).ParseGlob("templates/*.html")
```

Reply:
Done!